src/extensions/timeseries/TimeWidgetNew/TimeWidgetNew.tsx

```diff
@@ -197,13 +201,47 @@ const plotMargins = {
     bottom: 100
 }
 
+const useTimeState = (args: {
+    numTimepoints: number,
+    maxTimeSpan: number,
+    currentTimepoint?: number | null,
+    onCurrentTimepointChanged?: (t: number | null) => void,
+    timeRange?: {min: number, max: number} | null,
+    onTimeRangeChanged?: (r: {min: number, max: number} | null) => void
+}): {timeState: TimeState, timeDispatch: React.Dispatch<TimeAction>} => {
+    const { numTimepoints, maxTimeSpan, currentTimepoint, onCurrentTimepointChanged, timeRange, onTimeRangeChanged } = args
+    const [timeState, setTimeState] = useState<TimeState>({timeRange: null, currentTime: null, numTimepoints, maxTimeSpan})
+    let newTimeState = {...timeState}
+    if (currentTimepoint !== undefined) {
+        newTimeState = {...newTimeState, currentTime: currentTimepoint}
+    }
+    if (timeRange !== undefined) {
+        newTimeState = {...newTimeState, timeRange}
+    }
```

Comment:
Is there a circumstance under which `currentTimepoint` or `timeRange` could be reset to `undefined` when they previously had a value that could be preserved? I guess a particular call to `useTimeState` might omit those optional parameters if they aren't meant to be changed?

Reply:
I think we can assume this woud not change because the parent child component relationship should be consistent for this. But I think handling this could certainly be improved.